### PR TITLE
Return optimize_results and covariance_results in Fit.run() 

### PR DESCRIPTION
**Pull request by QRemy**
Modify Fit.run() to return a dict containing optimize_results and covariance_results.

### CODE REVIEW

File: docs/tutorials/analysis/time/light_curve_simulation.ipynb [hunk 1]
```diff
@@ -431,7 +431,25 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.0"
+   "version": "3.7.3"
+  },
```
Comment (adonath) [inline docs/tutorials/analysis/time/light_curve_simulation.ipynb:435]: Can you check your version of `nbconvert`? Like:
```
import nbconvert
print(nbconvert.__version__)
```
I think in more recent versions this meta data is not present anymore...

Comment (QRemy) [reply docs/tutorials/analysis/time/light_curve_simulation.ipynb:435]: ok upgraded to 6.1.0

**Comment by adonath** [comment]
Thanks @QRemy, I fixed the merge conflict caused by merging #3406. Once CI is green I'll go ahead and merge. Instead of the `dict` we could also consider actually renaming the current `FitResult` to e.g. `FitStepResult` and create a new `FitResult` object which bundles the results from the different steps.  But this only becomes useful if we add custom methods e.g. `FitResult.__str__` and / or `FitResult.write()`, it will be a separate PR anyway...

**Review by adonath — APPROVED** [review]
Thanks @QRemy!

**Pull request merged**